cola/widgets/branch.py

```diff
@@ -0,0 +1,520 @@
+        """Convert a list of delimited strings to a nested tree dict"""
+        result = {}
+        for item in list_branches:
+            tree = result
+            for part in item.split(separator_char):
+                tree = tree.setdefault(part, {})
+
+        return result
+
+    @staticmethod
+    def create_top_level_item(name, dict_children, child_icon):
+        """Create a top level tree item and its children """
+
+        def create_children(grouped_branches):
+            """Create children items for a tree item"""
+            result = []
+            for key in grouped_branches.keys():
+                item = BranchTreeWidgetItem(key, child_icon)
+                item.addChildren(create_children(grouped_branches[key]))
+
+                if item.childCount() > 0:
+                    item.setIcon(0, icons.ellipsis())
+
+                result.append(item)
+
+            result.sort(key=lambda x: x)
+
+            return result
+
+        branch = BranchTreeWidgetItem(name, icons.ellipsis())
+        branch.addChildren(create_children(dict_children))
+
+        return branch
+
+    @staticmethod
+    def get_root(item):
+        """Returns top level item from an item"""
+        parents = [item]
+        parent = item.parent()
+
+        while parent is not None:
+            parents.append(parent)
+            parent = parent.parent()
+
+        return parents[len(parents) - 1]
+
+    @staticmethod
+    def get_full_name(item, join_char):
+        """Returns item full name generated by iterating over
+        parents and joining their names with 'join_char'"""
+        parents = [item.name]
+        parent = item.parent()
+
+        while parent is not None:
+            parents.append(parent.name)
+            parent = parent.parent()
+
+        result = join_char.join(reversed(parents))
+
+        return result[result.find(join_char) + 1:]
+
+    @staticmethod
+    def expand_from_item(item):
+        """Expand tree parents from item"""
+        parent = item.parent()
+
+        while parent is not None:
+            parent.setExpanded(True)
+            parent = parent.parent()
+
+    def load_state(self, item, state):
+        """Load expanded items from a dict"""
+        if len(state.keys()) > 0:
+            item.setExpanded(True)
+
+        for i in range(item.childCount()):
+            child = item.child(i)
+            if child.name in state:
+                self.load_state(child, state[child.name])
+
+    def save_state(self, item):
+        """Save expanded items in a dict"""
+        result = {item.name: {}}
+
+        if item.isExpanded():
+            for i in range(item.childCount()):
+                child = item.child(i)
+                result[item.name].update(self.save_state(child))
+
+        return result
+
+
+class GitHelper(object):
+    def __init__(self, git):
+        self.git = git
+
+    def log(self, origin, args):
+        return self.git.log(origin, *args)
```

Comment:
I forgot to mention about the argument passing.. if you allow passing through both `*args` and `**kwargs` then you can let the Git class handle the option vs. argument stuff for you.

By passing the `--oneline` flag as `oneline=True` to the `log()` method lets the class handle the option vs. argument stuff.  That's probably easier to handle from the caller rather than trying to rearrange how the flags are passed.